src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1598,3 +1598,39 @@
   [custom-oid-mapping-file :- (schema/maybe schema/Str)]
   (merge (get-custom-oid-mappings custom-oid-mapping-file)
          (clojure.set/map-invert puppet-short-names)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; CA status info
+
+(schema/defn ca-expiration-dates
+  "Returns of a map of subject names of certs in the CA bundle
+  to their expiration dates."
+  [ca-cert-file :- schema/Str]
+  (let [ca-cert-bundle (utils/pem->certs ca-cert-file)]
+    (reduce (fn [cert-expirations cert]
+              (assoc cert-expirations
+                ;; Is this what we want to key off of?
+                (-> cert
+                    (.getSubjectDN)
```

Comment:
I think so, this will give something more like "Puppet CA: foo.bar.baz" in actual deployments

Reply:
From a customer facing standpoint I think that makes a lot of sense.  Works for me.

Reply:
Yeah that's what I saw in testing, I'd like @jonathannewman to confirm that's a useful key.